File: internal/domain/sync.go
package domain

import "time"

type SyncJob struct {
	ID               string     `json:"id"`
	WalletID         string     `json:"wallet_id"`
	Backend          string     `json:"backend"`
	Status           string     `json:"status"` // pending | running | done | failed
	AddressesScanned int        `json:"addresses_scanned"`
	TxFound          int        `json:"tx_found"`
	ErrorMsg         string     `json:"error_message,omitempty"`
	StartedAt        time.Time  `json:"started_at"`
	FinishedAt       *time.Time `json:"finished_at,omitempty"`
}

type SyncState struct {
	WalletID        string    `json:"wallet_id"`
	LastSyncedAt    time.Time `json:"last_synced_at"`
	ReceiveGapStart int       `json:"receive_gap_start"`
	ChangeGapStart  int       `json:"change_gap_start"`
	BlockHeight     int64     `json:"block_height"`
}

File: web/src/api/sync.ts
import { apiFetch } from './client'

export interface SyncJob {
  id: string
  wallet_id: string
  backend: string
  status: 'pending' | 'running' | 'done' | 'failed'
  addresses_scanned: number
  tx_found: number
  error_message?: string
  started_at: string
  finished_at?: string
}

export function startSync(walletID: string): Promise<{ job_id: string }> {
  return apiFetch(`/wallets/${walletID}/sync`, { method: 'POST' })
}

export function getSyncJob(jobID: string): Promise<SyncJob> {
  return apiFetch(`/sync-jobs/${jobID}`)
}

export function listSyncJobs(walletID: string): Promise<SyncJob[]> {
  return apiFetch(`/wallets/${walletID}/sync-jobs`)
}
